src/ui/suspense/sourceCaches.ts

```diff
@@ -0,0 +1,72 @@
+import type { SymbolDeclarations } from "devtools/client/debugger/src/reducers/ast";
+import { createGenericCache } from "replay-next/src/suspense/createGenericCache";
+import {
+  StreamingSourceContents,
+  getStreamingSourceContentsHelper,
+} from "replay-next/src/suspense/SourcesCache";
+import { ReplayClientInterface } from "shared/client/types";
+
+export const {
+  getValueAsync: getSourceContentsAsync,
+  getValueSuspense: getSourceContentsSuspense,
+  getStatus: getSourceContentsStatus,
+} = createGenericCache<
+  [replayClient: ReplayClientInterface, sourceId: string],
+  StreamingSourceContents | undefined
+>(
+  "sourceContentsCache",
+  async (replayClient, sourceId) => {
+    const res = await getStreamingSourceContentsHelper(replayClient, sourceId);
+    if (res) {
+      // Ensure that follow-on caches have the entire text available
+      const sourceContents = await res.resolver;
+      return sourceContents;
+    }
+  },
+  (replayClient, sourceId) => sourceId
+);
```

Comment:
If we need a wrapper around the streaming API, I think it would be nicer to co-locate it with the rest of the source cache (rather than string it out in multiple files): `packages/replay-next/src/suspense/SourcesCache`